rules/xcodeproj.bzl

```diff
@@ -442,18 +443,20 @@ def _populate_xcodeproj_targets_and_schemes(ctx, targets, src_dot_dots):
             "CLANG_ENABLE_OBJC_ARC": "YES",
         }
 
-        # Just like framework, we need to have absolute path to the hmap files
-        # so that Objc files can correctly import headers
-        target_settings["HEADER_SEARCH_PATHS"] = _joined_header_search_paths(
-            target_info.hmap_paths.to_list(),
-        )
+        all_hmaps = []
+        for at in all_transitive_targets:
+          if at.name == target_name:
```

Comment:
this feels a tad weird to me: can you drop a comment in here explaining what this is doing?